### DISCUSSION TITLE
More features for Ghostery

### DISCUSSION THREAD

**Discussion by 2br-2b** [ghostery/user-agent-desktop#495] [Ideas] [answered]
Hello!

I want to share this browser with my friends, but there's a small problem - not all of them are nerds like me, so it can be hard to convince them to switch to Ghostery. With most other browsers, there's a built-in marketing pitch:
- Chrome: most popular web browser
- Edge: fast and efficient, built-in by default to Windows
- Firefox: most popular non-chrome browser, Pocket
- Brave: Google Chrome without trackers, Brave Rewards, built-in ad blocking
- Opera: Opera Flow, built-in vpn, mouse gestures
- Vivaldi: complete customization
- Browsh: a browser that runs completely in your terminal

Of course, these are not these browsers' only selling points, but these are the main ones I'd use if I were to pitch the browsers to a friend

Right now, the pitch for Ghostery, according to the Readme, is:
- The Ghostery Privacy Blocker extension is included and enabled by default.
- Ghostery privacy search included.
- Firefox settings (telemetry etc) tuned for maximum privacy

These are all good things, but these aren't necessarily things that the average user might care about:
- Not all of my friends care enough about privacy to switch their browser
- Many people like using Google
- Unfortunately, this doesn't make too much of a difference to an end user.

Don't get me wrong - as a nerd, I love all of these changes, and I think they're great! The problem comes up trying to explain these to an average Joe and trying to get them to care about it. It would be hard for me to pitch these to another user as multiple reasons to switch - they'd see it as one reason, that the browser is more private.

Listening to GhostStories, it seems like you plan to keep support for Manifest v2 even if Firefox were to drop this ([they have no plans to do this now](https://blog.mozilla.org/addons/2019/09/03/mozillas-manifest-v3-faq/), but you never know about the future). Just supporting ad blockers and tracker blockers will be a huge boost when Google drops support. I bet there will be a large influx of users to other browsers from Google Chrome after manifest v2 is no longer supported. I, as someone who isn't associated with Ghostery, personally think Ghostery Browser's goal should be (at least in part) to have a good alternative browser ready to catch this wave of new users.

What could set Ghostery apart is having features which come as addons and extensions in other browsers built into Ghostery. For example, I used Opera for some time because of its flow, mouse/rocker gestures, and vpn. Even though these could be added into the browser via extensions, having them built in made the features seem much smoother in experience. It also helped that their settings pages were integrated into the browser's main settings page, making the whole process feel smoother and more integrated. It made me feel like a pro user having these features, and I wouldn't have gotten that feeling or even tried out these features unless they had been built in.

Obviously, there will be a tradeoff for adding extra features - as more features get added, it will take more processing power, thus slowing down the browser. However, even if a few of these were added, they could be huge game changers for some people and could be enough to convince some of those who don't otherwise care about privacy to switch to Ghostery.

While similar functionality could be added by #391, there's a *huge* difference to an end user if this feature is built-in instead of needing to go to the AMO, read the addon's description, press install, and even then not treating these features as core parts of the browser, but rather as add ons.

---

Some example features (some of which I've already added issues for) are:

## Progressive Web App Support (#467)
This is currently not implemented by Firefox and they have ended any plans of implementing this, but if this was even partially supported, it would be a big leap over Firefox towards matching Chromium's functionality

## Custom Search Engine Support (#470)
Add support to add custom search engines that don't support OpenSearch. Again, necessary to catch up with Chromium

## Website Dark Mode
For websites that don't natively support dark mode, extensions like [Dark Reader](https://addons.mozilla.org/en-US/firefox/addon/darkreader) change websites' css to provide dark mode for these websites.

## Mouse/Rocker Gestures
This was one of the huge differentiators for me, and one of the reasons I stayed with Opera for so long. It's hard to explain, and it doesn't really make sense until you try it yourself, but Rocker Gestures are a *HUGE* productivity booster, [Here is Opera's help page](https://help.opera.com/en/latest/shortcuts/#mouseGestures), but it doesn't explain that much. https://www.practicalecommerce.com/9-Mouse-Gestures-For-the-Opera-Browser better explains what these are. Some extensions that replicate this functionality in Firefox are [Gesturefy](https://addons.mozilla.org/en-US/firefox/addon/gesturefy) and [smartUp Gestures](https://addons.mozilla.org/en-US/firefox/addon/smartup/). After using mouse gestures for almost two years, it's nearly impossible for me to go back to using the web without them!

As a note, these work best with an external mouse.

## Local CDN
[Decentraleyes](https://addons.mozilla.org/en-US/firefox/addon/decentraleyes/) intercepts calls to Google APIs and instead has the files stored locally. By doing this, it 1) protects users' privacy by not pinging Google every time one of these pages is loaded, and 2) saves the bandwidth that would have otherwise been used to load these remote contents

## Remove Trackers from URLs
[ClearURLs](https://addons.mozilla.org/en-US/firefox/addon/clearurls/) removes trackers from urls. For example, it shortens `https://www.amazon.com/dp/exampleProduct/ref=sxin_0_pb?__mk_de_DE=ÅMÅŽÕÑ&keywords=tea&pd_rd_i=exampleProduct&pd_rd_r=8d39e4cd-1e4f-43db-b6e7-72e969a84aa5&pd_rd_w=1pcKM&pd_rd_wg=hYrNl&pf_rd_p=50bbfd25-5ef7-41a2-68d6-74d854b30e30&pf_rd_r=0GMWD0YYKA7XFGX55ADP&qid=1517757263&rnid=2914120011` to `https://www.amazon.com/dp/exampleProduct` to remove any metadata or UUIDs stored in the link. This may already be done by the Ghostery extension, but I thought I'd list it here just in case.

## Productivity Mode
Add some combination of [LeechBlock](https://addons.mozilla.org/en-US/firefox/addon/leechblock-ng/) and other addons to disable certain websites when you're working and to keep you on track. Alternatively, add something like [Vivaldi's break mode](https://www.youtube.com/watch?v=nK7H0OSx3yc) to temporarily block internet access on Ghostery

---

To summarize, I think adding additional features and additional talking points could be a huge boost to Ghostery's popularity. Since many of these features are implemented by open source addons, most of the heavy lifting is done, and it would be possible to work with the creators of the extensions to implement them into Ghostery.

Just looking at [Opera's homepage](https://www.opera.com/) shows how much of a difference these features can make. Even if they don't add much functionality, adding these features could be used for a marketing push, like what Brave did when [it integrated ipfs](https://brave.com/brave-integrates-ipfs/).

I'm sure there's a whole lot of behind-the-scenes changes going on with Ghostery (there's 74 open issues right now, after all), but in the long term, I think these could make a huge difference in setting Ghostery apart from its competitors. I'm also sure you're having these discussions internally, but I just wanted to give my feedback and help in any way that I can. The more browsers, the better, and more competition will force more mainstream browsers to continue to innovate so that they don't lose their market share.

If you prefer, I can add each of these ideas as a separate discussion or issue - Github's Discussions are still in beta, and I don't know if you would want these as separate issues, discussions, or something else, and I didn't want to spam y'all with a bunch of different discussions until making sure y'all were okay with it.

Anyway, thank you for your time, and I hope to help contribute to the Ghostery browser soon! I'm having some compiler issues, but I hope I'll have them resolved soon.

**Comment by tillhouse** [comment, marked as answer]
Hey @2br-2b, thanks for the awesome suggestions!  To your point, we're definitely looking for new ways to further differentiate Ghostery Dawn, and building out our own layer of unique services by leveraging the modular extension infrastructure exactly how we hope to do it quickly.  We plan to bring some of our own unique extensions to Dawn - some new and some existing.  This could include a VPN as you suggested, our [Ghostery Insights](https://www.ghostery.com/insights/) extension, or a new extension related to [Whotracks.me](https://whotracks.me/), but of course we can and should consider features that aren't strictly privacy-related.  We'll take your ideas into consideration and try to get some of these things live and in Dawn soon!

**Reply by 2br-2b** [reply]
Thanks for listening! I realized after posting this that this feedback probably should have been entered on y'all's survey rather than on Github Discussions. I had put something similar to this in the survey, but kept fleshing out the idea in my mind over the next couple of days, and I felt like it would be redundant to fill out the survey twice.

I also realized that y'all already must have a roadmap planned for the browser. I just hope that I can contribute, and am hoping to help contribute to improving Ghostery Browser, both in my own ways and in the ways y'all have planned! This is just something I've been mulling over for the past few weeks as I've been thinking about how I personally would go about fighting the browser wars.

I really like Ghostery Browser, and I'm working on setting up my dev environment so that I can help to implement some of these features (both the ones that I suggested and the ones y'all are working on). I look forward to helping Ghostery Browser to grow, and can't wait to see what happens as it evolves!
